test/krew_test.go

```diff
@@ -36,13 +58,117 @@ func TestKrewInstall(t *testing.T) {
 	test.Call(validPlugin, "--help").RunOrFail()
 }
 
-func TestKrewHelp(t *testing.T) {
+func TestKrewUninstall(t *testing.T) {
 	skipShort(t)
 
 	test, cleanup := krew.NewTest(t)
 	defer cleanup()
 
-	test.Krew("help").RunOrFail()
+	test.WithIndex().Krew("install", validPlugin).RunOrFailOutput()
+	test.Krew("uninstall", validPlugin).RunOrFailOutput()
+	if err := test.Call(validPlugin, "--help").Run(); err == nil {
+		t.Errorf("Expected the plugin to be uninstalled")
+	}
+}
+
+func TestKrewSearchAll(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	output := test.WithIndex().Krew("search").RunOrFailOutput()
+	if plugins := lines(output); len(plugins) < 10 {
+		// the first line is the header
+		t.Errorf("Expected at least %d plugins", len(plugins)-1)
+	}
+}
+
+func TestKrewSearchOne(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	plugins := lines(test.WithIndex().Krew("search", "krew").RunOrFailOutput())
+	if len(plugins) < 2 {
+		t.Errorf("Expected krew to be a valid plugin")
+	}
+	if !strings.HasPrefix(plugins[1], "krew ") {
+		t.Errorf("The first match should be krew")
+	}
+}
+
+func TestKrewInfo(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	output := test.WithIndex().Krew("info", validPlugin).RunOrFailOutput()
+	if !strings.HasPrefix(string(output), "NAME: "+validPlugin) {
+		t.Errorf("The info output should begin with the name header")
+	}
+}
+
+func TestKrewList(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	initialList := test.WithIndex().Krew("list").RunOrFailOutput()
+	test.Krew("install", validPlugin).RunOrFail()
+	eventualList := test.Krew("list").RunOrFailOutput()
+
+	if len(lines(eventualList))-len(lines(initialList)) != 1 {
+		t.Logf("initial list: %q\n", initialList)
+		t.Logf("eventual list: %q\n", eventualList)
+		t.Errorf("The list of installed plugins should grow by one when a single plugin is installed")
+	}
+}
+
+func TestKrewVersion(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	output := test.Krew("version").RunOrFailOutput()
+
+	requiredSubstrings := []string{
+		"IsPlugin",
+		fmt.Sprintf("BasePath        %s", test.Root()),
+		"ExecutedVersion",
+		"GitTag",
+		"GitCommit",
+		"IndexURI        https://github.com/kubernetes-sigs/krew-index.git",
```

Comment:
I feel like we'll easily break these spaces. :) 
I'm not sure if this whole thing is necessary. I currently only care about exitcode=0. :)

Reply:
I think it's worth a lot to know if something is present in the output. Do you like a regex-based assertion better?

Reply:
I think anything is fine.